sdk/keyvault/keyvault-certificates/review/keyvault-certificates.api.md

```diff
@@ -369,9 +366,8 @@ export type MergeCertificateOptions = coreHttp.OperationOptions;
 
 // @public
 export interface ParsedKeyVaultCertificatesIdentifier {
```

Comment:
Can we consider shortening the interface name? We use `Id` instead of `Identifier` throughout the package, so at the very least I would expect this to be `ParsedKeyVaultCertificatesId`.

After that I wonder if the KeyVault term is needed in this case. If not, then this can be further shortened to `ParsedCertificatesId`.

Also, am guessing this is tied to a single certificate? In which case we can remove the plural form and go with `ParsedCertificateId`. 

Same for keys and secrets

Reply:
Sure! Let's get Heath's thumbs up first @heaths ^

Reply:
My feedback regarding Id vs Identifier was specific to JS and not that we should make this change to all languages.

The rest of the feedback on removing the term "KeyVault" from the name and change `Certificates` to `Certificate` was language agnostic

Reply:
The type that has these IDs in the Key Vault libraries for TS is called: [KeyVaultCertificate](https://github.com/Azure/azure-sdk-for-js/blob/master/sdk/keyvault/keyvault-certificates/src/certificatesModels.ts#L186), and the property is called `id`, so I would personally agree to change this to `ParsedKeyVaultCertificateId`.

In fact, I'll do that change and request another review. If you both agree that `ParsedCertificateId` is better, then I'll change it to that.

Reply:
Updated: https://github.com/Azure/azure-sdk-for-js/pull/10736/commits/b43e07aa25f3f1b6e40b5cd16c3a7376cc9297e6

Reply:
I reviewed it a couple of times after doing the change, but please let me know if I missed something.

I didn't change the "Identifier" string from the tests, because using "id" for variables (instead of "identifier", which is what the tests have at the moment) seemed too small to represent something valuable. For the tests variables, "identifier" felt more appropriate to me.

Reply:
I understand that `KeyVaultCertificate` has the term KeyVault in it, but then `CertificatePolicy` does not. So, where do we draw the line as to which interfaces get to have `KeyVault` in it and which should not?

The language agnostic motivation for the guideline on having the service name as a prefix is to allow users to know which type/classes is coming from which library. 

It is more important for words which are generic and therefore the prefix provides context, and less important for something that is already specific enough that one would not mistake it to be a generic thing from another library.

It is more important for classes that one would instantiate and less important for interfaces of output results which no JS user would use and a small set of TS user would use.

Reply:
I generally don't like to argue too much about names, but I think I can point this discussion towards the people that might like to argue about names more than me.

However, I'll express my opinion first, since it's not like I don't have one.

Part of the reasoning of having a prefix for keys, secrets and certificates, is that they're not universal keys, secrets or certificates, they're very specific to Azure Key Vault. That same logic applies to this identifier, in my opinion. An identifier can be anything, and a certificate identifier almost just as much. However, a _Key Vault certificate identifier_ can only mean one thing.

Here's the other discussion about prefixes where we need to coordinate as a team. Your feedback is appreciated there too! https://github.com/Azure/azure-sdk-for-js/pull/10815#discussion_r477240112

Reply:
Since I don't like to argue too much about names, [I did the update: I removed `KeyVault` from these types and functions](https://github.com/Azure/azure-sdk-for-js/pull/10736/commits/813c3f884fa4f8e9e71b329a0bdf43d882afb679).

@ramya-rao-a if you approve this PR, I'll merge it.

Later, if the architects decide to rename this, I'm ok changing this to have a different name.

The reason I don't like naming discussions is because, for me, they're exactly like linting discussions. Naming and linting are important, and they arguably can be supported by entire fields of study (as in, design principles, etymology, mnemonics, sociology, etc). To me, the exact proper name is hard to find, and below that ideal, many acceptable names are easy to find. The proposals here are all acceptable to me. Instead of having to say that I prefer one than another from some gut feeling, I'd rather have concrete guidelines about naming, just like how we have concrete guidelines about linting. To get guidelines, we need to meet with the architects until we're able to write something down that we all agree. This is why I'm pointing you to the other thread.

Reply:
The prefix was to match the type names we had to use, i.e. `KeyVaultCertificateIdentifier` (.NET name) goes with `KeyVaultCertificate`. Same for keys and secrets.